common/build-style/gem.sh

```diff
@@ -0,0 +1,125 @@
+RbConfig::MAKEFILE_CONFIG['LD'] = ENV['LD'] if ENV['LD']
+RbConfig::MAKEFILE_CONFIG['CFLAGS'] = ENV['CFLAGS'] if ENV['CFLAGS']
+RbConfig::MAKEFILE_CONFIG['CXXFLAGS'] = ENV['CXXFLAGS'] if ENV['CXXFLAGS']
+EOF
+
+		# Patch all instances of extconf that use create_makefile
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep create_makefile $f || continue
+				# Create a backup which we will restore later
+				cp $f ${f}.orig
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
+	fi
+
+	$gem_cmd build ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Restrore previous extconf.rb which we ship.
+		for f in $(find . -type f -name 'extconf.rb.orig'); do
+			mv $f ${f%.*}
+		done
+	fi
+}
+
+do_install() {
+	: ${gem_cmd:=gem}
+
+	local _GEMDIR _TARGET_PLATFORM _HOST_PLATFORM _HOST_EXT_DIR _TARGET_EXT_DIR
+
+	_GEMDIR="$($gem_cmd env gemdir)"
+
+	if [ "$CROSS_BUILD" ]; then
+		# Convert to the platforms used in our ruby packages
+		case "$XBPS_TARGET_MACHINE" in
+			x86_64) _TARGET_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _TARGET_PLATFORM=x86_64-linux-musl ;;
+			i686) _TARGET_PLATFORM=i686-linux ;;
+			i686-musl) _TARGET_PLATFORM=i686-linux-musl ;;
+			armv7l) _TARGET_PLATFORM=/armv7l-linux-eabihf ;;
+			armv6l) _TARGET_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _TARGET_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _TARGET_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _TARGET_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _TARGET_PLATFORM=aarch64-linux-musl ;;
+		esac
+
+		case "$XBPS_MACHINE" in
+			x86_64) _HOST_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _HOST_PLATFORM=x86_64-linux-musl ;;
+			i686) _HOST_PLATFORM=i686-linux ;;
+			i686-musl) _HOST_PLATFORM=i686-linux-musl ;;
+			armv7l) _HOST_PLATFORM=/armv7l-linux-eabihf ;;
+			armv6l) _HOST_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _HOST_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _HOST_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _HOST_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _HOST_PLATFORM=aarch64-linux-musl ;;
+		esac
+	fi
+
+	$gem_cmd install \
+		--local \
+		--install-dir ${DESTDIR}/${_GEMDIR} \
+		--bindir ${DESTDIR}/usr/bin \
+		--ignore-dependencies \
+		--verbose \
+		--no-document \
+		${pkgname#ruby-}-${version}.gem
+
+	# Useless files
+	rm -rf ${DESTDIR}/${_GEMDIR}/cache
+	rm -rf ${DESTDIR}/${_GEMDIR}/gems/${pkgname#ruby-}-${version}/{test,LICENSE,README,*.adoc}
+
+	if [ "$CROSS_BUILD" ]; then
+		# Path to the extensions on a package, ruby installs against the platform
+		# of the host, so we have to move them to the correct place
+		_HOST_EXT_DIR="${DESTDIR}/${_GEMDIR}/extensions/${_HOST_PLATFORM}"
+		_TARGET_EXT_DIR="${DESTDIR}/${_GEMDIR}/extensions/${_TARGET_PLATFORM}"
+
+		if [ -d ${_HOST_EXT_DIR} -a "${_HOST_EXT_DIR}" != "${_TARGET_EXT_DIR}" ]; then
+			mv ${_HOST_EXT_DIR} ${_TARGET_EXT_DIR}
+		fi
```

Comment:
When would `_HOST_EXT_DIR` be `_TARGET_EXT_DIR` during cross compiling?

Reply:
When ?

Reply:
How would these two ever match during cross.